### PR TITLE
Bump pyo3

### LINKED ISSUE #4852: Pyo3 replace `parse_kwargs` with `pyo3::signature`
Previously, we use python dictionnary to pass argument in any order:

```rust
#[pymethods]
impl RealmRoleCertificate {
    #[new]
    #[pyo3(signature = (**py_kwargs))]
    fn new(py_kwargs: Option<&PyDict>) -> PyResult<Self> {
        crate::binding_utils::parse_kwargs!(
            py_kwargs,
            [author: Option<DeviceID>, "author"],
            [timestamp: DateTime, "timestamp"],
            [realm_id: RealmID, "realm_id"],
            [user_id: UserID, "user_id"],
            [role: Option<RealmRole>, "role"],
        );

        Ok(Self(Arc::new(
            libparsec::low_level::types::RealmRoleCertificate {
                timestamp: timestamp.0,
                author: match author {
                    Some(device_id) => CertificateSignerOwned::User(device_id.0),
                    None => CertificateSignerOwned::Root,
                },
                realm_id: realm_id.0,
                user_id: user_id.0,
                role: role.map(|x| x.0),
            },
        )))
    }
    // ...
}
```

This required having boilerplate code (`parse_kwargs`) and could be replaced by `pyo3::signature`

Could become:

```rust
#[pymethods]
impl RealmRoleCertificate {
    #[new]
    #[pyo3(signature = (*, author, timestamp, realm_id, role))]
    fn new(author: Option<DeviceID>, timestamp: DateTime, realm_id: RealmID, role: Option<RealmRole>) -> PyResult<Self> {
        Ok(Self(Arc::new(
            libparsec::low_level::types::RealmRoleCertificate {
                timestamp: timestamp.0,
                author: match author {
                    Some(device_id) => CertificateSignerOwned::User(device_id.0),
                    None => CertificateSignerOwned::Root,
                },
                realm_id: realm_id.0,
                user_id: user_id.0,
                role: role.map(|x| x.0),
            },
        )))
    }
    // ...
}
```

> Note the `*` at the start of `signature = (...`, it indicate the following argument are keyword only.

## TODOs

- [ ] Pyo3 replace custom macro `parse_kwargs` with `pyo3::signature`.
- [ ] Pyo3 replace custom macro `parse_kwargs_optional` with `pyo3::signature`.

### PR DESCRIPTION
**Pull request by TimeEngineer**
## Describe your changes

Bump `pyo3` (#[args] is now deprecated, and arguments following an Option are not allowed) & remove `parse_kwargs`
closes #4852 

## Checklist

Before you submit this pull request, please make sure to:

- [x] Keep changes in the pull request as small as possible
   
- [x] Ensure the commit history is sanitized
   
- [x] Give a meaningful title to your PR
- [x] Describe your changes
   
- [x] Link any related issue in the description
   
- [x] Link any dependent pull request in the description

If this PR adds a new feature or fixes a bug:

- [ ] I have added or updated relevant tests

If this PR modifies installation or updates packages:

- [ ] I have checked that installation works on all platforms (Windows, Linux, Mac ...)
    

- [ ] I have updated related documentations (installation commands, fix to specific problems ...)

If this PR is related to GUI:

- [ ] I have checked that GUIs affected by my changes are working properly.

If this PR affects the user:

- [ ] I have created a news fragment
   
- [ ] I have updated the user documentation

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -114,7 +114,7 @@ paste = { version = "1.0.12", default-features = false }
 percent-encoding = { version = "2.2.0", default-features = false }
 pretty_assertions = { version = "1.3.0", default-features = false }
 proc-macro2 = { version = "1.0.56", default-features = false }
-pyo3 = { version = "0.17.3", default-features = false }
+pyo3 = { version = "0.19.1", default-features = false }
```
Comment (FirelightFlagboy) [inline Cargo.toml:117]: FYI: A PR is already openned for that #4853

File: server/src/addrs.rs [hunk 2]
```diff
@@ -57,7 +58,7 @@ impl BackendAddr {
         self.0.to_url().to_string()
     }
 
-    #[args(path = "\"\"")]
+    #[pyo3(signature = (path = "\"\""))]
```
Comment (FirelightFlagboy) [inline server/src/addrs.rs:61]:
```suggestion
    #[pyo3(signature = (path = ""))]
```

The default value is `""` (empty string) it's different from `'""'` (string containing `""`)

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]
Minor change required for default string

File: Cargo.toml (hunk 1 above)
Comment (TimeEngineer) [reply Cargo.toml:117]: 👍 I didn't know

File: server/src/addrs.rs (hunk 2 above)
Comment (TimeEngineer) [reply server/src/addrs.rs]: Missed, done

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**